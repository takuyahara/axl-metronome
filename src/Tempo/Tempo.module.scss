@mixin common-style {
    color: orange;
    font-size: 60px;
    font-family: "DIN Alternate";
    user-select: none;
    -webkit-touch-callout: none;
    -webkit-user-select: none;
}
.from .tempoValue,
.to .tempoValue {
    position: fixed;
}
.from {
    @include common-style;
    width: 100%;
    height: 50%;
    background: rgb(0, 0, 0);
    background: -moz-linear-gradient(top, rgb(0, 0, 0) 0%, rgb(40, 40, 40) 100%);
    background: -webkit-linear-gradient(top, rgb(0, 0, 0) 0%, rgb(40, 40, 40) 100%);
    background: linear-gradient(to bottom, rgb(0, 0, 0) 0%, rgb(40, 40, 40) 100%);
    display: flex;
    justify-content: center;
    align-items: center;
}
.to {
    @include common-style;
    width: 100%;
    height: 50%;
    background: rgb(40, 40, 40);
    background: -moz-linear-gradient(top, rgb(40, 40, 40) 0%, rgb(0, 0, 0) 100%);
    background: -webkit-linear-gradient(top, rgb(40, 40, 40) 0%, rgb(0, 0, 0) 100%);
    background: linear-gradient(to bottom, rgb(40, 40, 40) 0%, rgb(0, 0, 0) 100%);
    display: flex;
    justify-content: center;
    align-items: center;
}
