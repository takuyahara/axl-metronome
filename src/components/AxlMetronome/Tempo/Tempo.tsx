import React, { Component, createRef, RefObject } from 'react';
import _ from 'lodash';
import style from './Tempo.module.scss';
import Utils, { ICssSelector } from 'utils/Utils';

interface IState {
  tempoCurr: number;
}
interface IProps {
  role: string;
  tempo: number;
  maxDelta: number;
  range: {
    from: number;
    to: number;
  };
  onChangeTempo?: (t: Tempo) => void;
  inheritedSelector?: ICssSelector;
}
interface IBCRSize {
  width: number;
  height: number;
}
class Tempo extends Component<IProps, IState> {
  // Initialize in init()
  private tempoPrev!: number;
  private role!: IProps['role'];
  private maxDelta!: IProps['maxDelta'];
  private range!: IProps['range'];
  private isTapped!: boolean;
  private distFrom!: number;
  private postChangeTempo!: (t: Tempo) => void;

  // Ref
  private refDiv: RefObject<HTMLDivElement>;

  public constructor(props: IProps) {
    super(props);
    this.state = {
      tempoCurr: props.tempo,
    };

    // Ref
    this.refDiv = createRef<HTMLDivElement>();

    // Method binding
    this.mouseMove = this.mouseMove.bind(this);
    this.mouseUp = this.mouseUp.bind(this);
    this.mouseDown = this.mouseDown.bind(this);

    // Local variables
    this.role = props.role;
    this.maxDelta = props.maxDelta;
    this.range = props.range;
    this.isTapped = false;
    this.distFrom = 0;
    this.tempoPrev = props.tempo;
    this.postChangeTempo = props.onChangeTempo !== undefined  ? props.onChangeTempo : () => {};
  }
  /* istanbul ignore next */
  private isPortrait(): boolean {
    // return ['portrait-primary', 'portrait-secondary'].includes(
    //   // tslint:disable-next-line:no-any
    //   ((screen as any).orientation || (screen as any).mozOrientation || (screen as any).msOrientation).type,
    // );
    return document.body.clientHeight > document.body.clientWidth;
  }
  private mouseMove(e: React.MouseEvent | React.TouchEvent): void {
    const isMouseEvent = e.type.substr(0, 5) === 'mouse';
    isMouseEvent && e.preventDefault();
    if (!this.isTapped) { return; }
    
    const tempoPrev = this.tempoPrev;
    const clientX = isMouseEvent ? (e as React.MouseEvent).clientX : (e as React.TouchEvent).changedTouches[0].clientX;
    const clientY = isMouseEvent ? (e as React.MouseEvent).clientY : (e as React.TouchEvent).changedTouches[0].clientY;
    const dist = this.isPortrait() ? this.getDist(this.distFrom, clientX) : this.getDist(clientY, this.distFrom);
    const newTempo = tempoPrev + parseInt(dist.toString());
    this.changeTempo(newTempo);
  }
  private mouseDown(e: React.MouseEvent | React.TouchEvent): void {
    const isMouseEvent = e.type.substr(0, 5) === 'mouse';
    isMouseEvent && e.preventDefault();

    const clientX = isMouseEvent ? (e as React.MouseEvent).clientX : (e as React.TouchEvent).changedTouches[0].clientX;
    const clientY = isMouseEvent ? (e as React.MouseEvent).clientY : (e as React.TouchEvent).changedTouches[0].clientY;
    this.distFrom = this.isPortrait() ? clientX : clientY;
    this.isTapped = true;
    this.tempoPrev = this.state.tempoCurr;
  }
  private mouseUp(e: React.MouseEvent | React.TouchEvent): void {
    const isMouseEvent = e.type.substr(0, 5) === 'mouse';
    isMouseEvent && e.preventDefault();

    this.isTapped = false;
    this.tempoPrev = this.state.tempoCurr;
  }
  /* istanbul ignore next */
  private getBCRSize(elem: HTMLElement): IBCRSize {
    // Excluded to be replaced in test.
    const { width, height } = elem.getBoundingClientRect();
    return {
      width,
      height,
    };
  }
  private normalizeTempo(tempo: number): number {
    let newTempo = tempo;
    if (newTempo < this.range.from) { 
      newTempo = this.range.from; 
    } else if (newTempo > this.range.to) { 
      newTempo = this.range.to; 
    }
    return newTempo;
  }
  public getTempo(): number {
    return this.state.tempoCurr;
  }
  public changeTempo(tempo: number, dontRunCallback?: boolean): void {
    const newTempo = this.normalizeTempo(tempo);
    
    const cbSetState = () => {
      if (!dontRunCallback) {
        const caller = this;
        this.postChangeTempo(caller);
      }
    };
    this.setState({
      tempoCurr: newTempo,
    }, cbSetState);
  }
  public getDist(distFrom: number, distTo: number): number {
    const refDiv = this.refDiv.current!;
    const { width, height } = this.getBCRSize(refDiv);
    const dist = distTo - distFrom;
    const length = this.isPortrait() ? width : height;
    const distNormalized = (dist / length) * this.maxDelta;
    return distNormalized;
  }
  public init(props: IProps): void {
    // State
    this.setState({
      tempoCurr: props.tempo,
    });
    
    // Local variables
    this.role = props.role;
    this.maxDelta = props.maxDelta;
    this.range = props.range;
    this.isTapped = false;
    this.distFrom = 0;
    this.tempoPrev = props.tempo;
    this.postChangeTempo = props.onChangeTempo !== undefined  ? props.onChangeTempo : () => {};
  }
  /* istanbul ignore next */
  public componentWillMount(): void {
    if (this.props.inheritedSelector) {
      Utils.inheritSelector(style, this.props.inheritedSelector!);
    }
  }
  /* istanbul ignore next */
  public componentWillReceiveProps(newProps: IProps): void {
    if (_.isEqual(this.props, newProps)) {
      return;
    }

    if (newProps.range.from > newProps.range.to) {
      throw new RangeError(`Range.from cannot be greater than Range.to.`);
    }
    if (newProps.tempo > newProps.range.to) {
      throw new RangeError(`Tempo cannot be greater than Range.to.`);
    }
    if (newProps.tempo < newProps.range.from) {
      throw new RangeError(`Tempo cannot be less than Range.from.`);
    }
    this.init(newProps);
  }
  public render(): React.ReactNode {
    return (
      <div 
        ref={this.refDiv} 
        className={style[this.role]} 
        data-testid={`temposelector-${this.role}`}
        onMouseDown={this.mouseDown} 
        onMouseUp={this.mouseUp} 
        onMouseMove={this.mouseMove} 
        onTouchStart={this.mouseDown} 
        onTouchEnd={this.mouseUp} 
        onTouchMove={this.mouseMove} 
      >
        <span>
          {this.state.tempoCurr}
        </span>
      </div>
    );
  }
}
export default Tempo;
