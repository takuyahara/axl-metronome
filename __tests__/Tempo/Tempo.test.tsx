import React from 'react';
import { configure, mount, ReactWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import Tempo from 'AxlMetronome/Tempo/Tempo';

interface ISpy {
  [s: string]: {
    [s: string]: sinon.SinonSpy;
  };
}

configure({ adapter: new Adapter() });
const sel = (id: string) => `[data-testid="${id}"]`;

let tempo: ReactWrapper;
let tempoInstance: Tempo;
let spy: ISpy = {};

// tslint:disable:no-any
describe('Tempo', () => {
  beforeEach(() => {
    // Sinon
    spy = {
      Tempo: {
        getDist: sinon.spy(Tempo.prototype, "getDist"),
        changeTempo: sinon.spy(Tempo.prototype, "changeTempo"),
        mouseDown: sinon.spy(Tempo.prototype as any, "mouseDown"),
        mouseMove: sinon.spy(Tempo.prototype as any, "mouseMove"),
        mouseUp: sinon.spy(Tempo.prototype as any, "mouseUp"),        
      },
    };
  });
  afterEach(() => {
    expect(tempo).toMatchSnapshot();
    tempo.unmount();
    sinon.restore();
  });

  describe('isPortrait = true', () => {
    beforeEach(() => {
      sinon.replace(Tempo.prototype as any, 'isPortrait', () => true);
      sinon.replace(Tempo.prototype as any, 'getBCRSize', () => ({
        width: 500,
        height: 800,
      }));
      
      // Enzyme
      tempo = mount(
        <Tempo 
          role="from" 
          tempo={80} 
          maxDelta={100} 
          range={{
            from: 50,
            to: 200,
          }} 
        />);
      tempoInstance = tempo.instance() as Tempo;
    
      // Sinon - reset
      sinon.resetHistory();
    });

    it('changes tempo', () => {
      expect(tempoInstance.getTempo()).toBe(80);
      tempoInstance.changeTempo(150);
      expect(tempoInstance.getTempo()).toBe(150);
    });

    it('normalize tempo to min when passed value is less than than min', () => {
      tempoInstance.changeTempo(30);
      expect(tempoInstance.getTempo()).toBe(50);
    });

    it('normalize tempo to max when passed value is more than than max', () => {
      tempoInstance.changeTempo(300);
      expect(tempoInstance.getTempo()).toBe(200);
    });

    describe('MouseEvent', () => {
      it('mouseMove and nothing happens', () => {
        tempo.find(sel('temposelector-from')).simulate('mousemove', {
          clientX: 300,
          clientY: 0,
        });

        expect(spy.Tempo.getDist.notCalled).toBe(true);
        expect(spy.Tempo.changeTempo.notCalled).toBe(true);
      });

      it('mouseDown', () => {
        tempo.find(sel('temposelector-from')).simulate('mousedown', {
          clientX: 100,
          clientY: 0,
        });
    
        expect((tempoInstance as any).distFrom).toBe(100);
        expect((tempoInstance as any).isTapped).toBe(true);
      });
    
      it('mouseDown => mouseMove', () => {
        tempo.find(sel('temposelector-from')).simulate('mousedown', {
          clientX: 100,
          clientY: 0,
        });
        tempo.find(sel('temposelector-from')).simulate('mousemove', {
          clientX: 300,
          clientY: 0,
        });

        expect(spy.Tempo.getDist.calledOnceWith(100, 300)).toBe(true);
        expect(spy.Tempo.getDist.returned(40)).toBe(true);
        expect(spy.Tempo.changeTempo.calledOnceWith(120)).toBe(true);
      });
    
      it('mouseDown => mouseMove => mouseUp', () => {
        tempo.find(sel('temposelector-from')).simulate('mousedown', {
          clientX: 100,
          clientY: 0,
        });
        tempo.find(sel('temposelector-from')).simulate('mousemove', {
          clientX: 300,
          clientY: 0,
        });
        tempo.find(sel('temposelector-from')).simulate('mouseup');

        expect((tempoInstance as any).isTapped).toBe(false);
      });
    });

    describe('TouchEvent', () => {
      it('touchMove and nothing happens', () => {
        tempo.find(sel('temposelector-from')).simulate('touchmove', {
          changedTouches: [
            {
              clientX: 300,
              clientY: 0,
            },
          ],
        });

        expect(spy.Tempo.getDist.notCalled).toBe(true);
        expect(spy.Tempo.changeTempo.notCalled).toBe(true);
      });

      it('touchStart', () => {
        tempo.find(sel('temposelector-from')).simulate('touchstart', {
          changedTouches: [
            {
              clientX: 100,
              clientY: 0,
            },
          ],
        });
    
        expect((tempoInstance as any).distFrom).toBe(100);
        expect((tempoInstance as any).isTapped).toBe(true);
      });

      it('touchStart => touchMove', () => {
        tempo.find(sel('temposelector-from')).simulate('touchstart', {
          changedTouches: [
            {
              clientX: 100,
              clientY: 0,
            },
          ],
        });
        tempo.find(sel('temposelector-from')).simulate('touchmove', {
          changedTouches: [
            {
              clientX: 300,
              clientY: 0,
            },
          ],
        });

        expect(spy.Tempo.getDist.calledOnceWith(100, 300)).toBe(true);
        expect(spy.Tempo.getDist.returned(40)).toBe(true);
        expect(spy.Tempo.changeTempo.calledOnceWith(120)).toBe(true);
      });

      it('touchStart => touchMove => touchEnd', () => {
        tempo.find(sel('temposelector-from')).simulate('touchstart', {
          changedTouches: [
            {
              clientX: 100,
              clientY: 0,
            },
          ],
        });
        tempo.find(sel('temposelector-from')).simulate('touchmove', {
          changedTouches: [
            {
              clientX: 300,
              clientY: 0,
            },
          ],
        });
        tempo.find(sel('temposelector-from')).simulate('touchend');

        expect((tempoInstance as any).isTapped).toBe(false);
      });
    });
  });

  describe('isPortrait = false', () => {
    beforeEach(() => {
      sinon.replace(Tempo.prototype as any, 'isPortrait', () => false);
      sinon.replace(Tempo.prototype as any, 'getBCRSize', () => ({
        width: 800,
        height: 500,
      }));
      
      // Enzyme
      tempo = mount(
        <Tempo 
          role="from" 
          tempo={80} 
          maxDelta={100} 
          range={{
            from: 50,
            to: 200,
          }} 
        />);
      tempoInstance = tempo.instance() as Tempo;
    
      // Sinon - reset
      sinon.resetHistory();
    });

    describe('MouseEvent', () => {
      it('mouseDown', () => {
        tempo.find(sel('temposelector-from')).simulate('mousedown', {
          clientX: 0,
          clientY: 300,
        });
    
        expect((tempoInstance as any).distFrom).toBe(300);
        expect((tempoInstance as any).isTapped).toBe(true);
      });
    
      it('mouseDown => mouseMove', () => {
        tempo.find(sel('temposelector-from')).simulate('mousedown', {
          clientX: 0,
          clientY: 300,
        });
        tempo.find(sel('temposelector-from')).simulate('mousemove', {
          clientX: 0,
          clientY: 100,
        });
        expect(spy.Tempo.getDist.calledOnceWith(100, 300)).toBe(true);
        expect(spy.Tempo.getDist.returned(40)).toBe(true);
        expect(spy.Tempo.changeTempo.calledOnceWith(120)).toBe(true);
      });
    
      it('mouseDown => mouseMove => mouseUp', () => {
        tempo.find(sel('temposelector-from')).simulate('mousedown', {
          clientX: 0,
          clientY: 300,
        });
        tempo.find(sel('temposelector-from')).simulate('mousemove', {
          clientX: 0,
          clientY: 100,
        });
        tempo.find(sel('temposelector-from')).simulate('mouseup');

        expect((tempoInstance as any).isTapped).toBe(false);
      });
    });

    describe('TouchEvent', () => {
      it('touchStart', () => {
        tempo.find(sel('temposelector-from')).simulate('touchstart', {
          changedTouches: [
            {
              clientX: 0,
              clientY: 300,
            },
          ],
        });
    
        expect((tempoInstance as any).distFrom).toBe(300);
        expect((tempoInstance as any).isTapped).toBe(true);
      });

      it('touchStart => touchMove', () => {
        tempo.find(sel('temposelector-from')).simulate('touchstart', {
          changedTouches: [
            {
              clientX: 0,
              clientY: 300,
            },
          ],
        });
        tempo.find(sel('temposelector-from')).simulate('touchmove', {
          changedTouches: [
            {
              clientX: 0,
              clientY: 100,
            },
          ],
        });

        expect(spy.Tempo.getDist.calledOnceWith(100, 300)).toBe(true);
        expect(spy.Tempo.getDist.returned(40)).toBe(true);
        expect(spy.Tempo.changeTempo.calledOnceWith(120)).toBe(true);
      });

      it('touchStart => touchMove => touchEnd', () => {
        tempo.find(sel('temposelector-from')).simulate('touchstart', {
          changedTouches: [
            {
              clientX: 0,
              clientY: 300,
            },
          ],
        });
        tempo.find(sel('temposelector-from')).simulate('touchmove', {
          changedTouches: [
            {
              clientX: 0,
              clientY: 100,
            },
          ],
        });
        tempo.find(sel('temposelector-from')).simulate('touchend');

        expect((tempoInstance as any).isTapped).toBe(false);
      });
    });
  });
});
