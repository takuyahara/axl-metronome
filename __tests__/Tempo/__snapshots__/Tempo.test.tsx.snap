// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tempo isPortrait = false MouseEvent mouseDown => mouseMove => mouseUp 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false MouseEvent mouseDown => mouseMove 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false MouseEvent mouseDown 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false MouseEvent toggleHandler() * 2 => mouseDown => mouseMove => mouseUp 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false MouseEvent toggleHandler() => mouseDown => mouseMove => mouseUp 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false TouchEvent toggleHandler() * 2 => touchStart => touchMove => touchEnd 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false TouchEvent toggleHandler() => touchStart => touchMove => touchEnd 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false TouchEvent touchStart => touchMove => touchEnd 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false TouchEvent touchStart => touchMove 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false TouchEvent touchStart 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false changes tempo 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false normalize tempo to max when passed value is more than than max 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false normalize tempo to min when passed value is less than than min 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = false returns current tempo 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true MouseEvent mouseDown => mouseMove => mouseUp 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true MouseEvent mouseDown => mouseMove 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true MouseEvent mouseDown 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true MouseEvent toggleHandler() * 2 => mouseDown => mouseMove => mouseUp 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true MouseEvent toggleHandler() => mouseDown => mouseMove => mouseUp 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true TouchEvent toggleHandler() * 2 => touchStart => touchMove => touchEnd 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true TouchEvent toggleHandler() => touchStart => touchMove => touchEnd 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true TouchEvent touchStart => touchMove => touchEnd 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true TouchEvent touchStart => touchMove 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      120
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true TouchEvent touchStart 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true changes tempo 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true normalize tempo to max when passed value is more than than max 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true normalize tempo to min when passed value is less than than min 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;

exports[`Tempo isPortrait = true returns current tempo 1`] = `
<Tempo
  isHandlerEnabled={true}
  maxDelta={100}
  range={
    Object {
      "from": 50,
      "to": 200,
    }
  }
  role="from"
  tempo={80}
>
  <div
    className="from"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="tempoValue"
    >
      80
    </span>
  </div>
</Tempo>
`;
